### PR TITLE
Add tables support for the rich text editor

### PR DESCRIPTION
**Pull request by vhuseinova-msft**
# What

- Add tables support for the rich text editor
- Fixed issue with screenshots for send box for stable (ui tests should be run for beta only as strings/icons are available for beta only)
Known issue: pasting table from Teams

# Why

# How Tested

Storybook

# Process & policy checklist

- [x] I have updated the project documentation to reflect my changes if necessary.
- [x] I have read the [CONTRIBUTING](https://github.com/Azure/communication-ui-library/blob/main/CONTRIBUTING.md) documentation.

**Is this a breaking change?**

- [ ] This change causes current functionality to break.

### CODE REVIEW

File: packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+  const { item, onClick, theme } = props;
+  const [column, setColumn] = React.useState(1);
+  const [row, setRow] = React.useState(1);
+
+  const updateSize = React.useCallback((target?: HTMLElement) => {
+    if (target !== undefined) {
+      const column = parseInt(target.dataset.column ?? '-1');
+      const row = parseInt(target.dataset.row ?? '-1');
+
+      if (column > 0 && column <= MaxColumns && row > 0 && row <= MaxRows) {
```
Comment (palatter) [inline packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx:37]: So there is no column = 0?

File: packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+import {
+  insertTableMenuCellButtonSelectedStyles,
+  insertTableMenuCellButtonStyles,
+  insertTableMenuFocusZone,
+  insertTableMenuTitleStyles
+} from '../../../styles/RichTextEditor.styles';
+import { ColumnRowReplaceString, createKey } from '../../../utils/RichTextTableUtils';
+
+// This file uses RoosterJS React package implementation with updates to UI components and styles.
+const MaxRows = 5;
```
Comment (palatter) [inline packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx:16]: Do you want make max rows/columns configurable?

File: packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx (hunk 1 above)
Comment (vhuseinova-msft) [reply packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx:37]: no, I've copied this file (almost everything) from RoosterJS and everything is started from 1. See `const items = React.useMemo(() => {` for more details.

I can update if you think we should start from 0

Comment (palatter) [reply packages/react-components/src/components/RichTextEditor/Buttons/Table/RichTextInsertTablePane.tsx:37]: We can leave as 1 if that is how rooster originally wrote it. Just thought it was strange the our range value above was set for -1 and then started at 1.

**Comment by palatter** [comment]
I notice file upload cards look different now?

image

**Comment by Leah-Xia-Microsoft** [comment]
Seems like in the Storybook, we cannot remove the files in the sendbox, thus we cannot send out the message to see the message content. 
image
Should we add the functionality to remove the files and/or have a separate section for file upload and one for the normal RTE sendbox?

**Comment by vhuseinova-msft** [comment]
> Seems like in the Storybook, we cannot remove the files in the sendbox, thus we cannot send out the message to see the message content. image Should we add the functionality to remove the files and/or have a separate section for file upload and one for the normal RTE sendbox?

cc @palatter could you take a look? It looks like it's something that was missed from the file upload functionality

**Comment by vhuseinova-msft** [comment]
> I notice file upload cards look different now?
> 
> image
> This is how they should look:
> 
> image
> This might unrelated to your PR though, let me double check on main right now.

Added missed theaming

**Comment by palatter** [comment]
> Seems like in the Storybook, we cannot remove the files in the sendbox, thus we cannot send out the message to see the message content. image Should we add the functionality to remove the files and/or have a separate section for file upload and one for the normal RTE sendbox?

**Review by palatter — APPROVED** [review]

**Review by Leah-Xia-Microsoft — APPROVED** [review]

**Pull request merged**